docs/arch/04-secrets-management.md

```diff
@@ -0,0 +1,183 @@
+# Secrets Management
+
+ToolHive provides a secrets management system for securely handling API keys, tokens, and other sensitive data needed by MCP servers.
+
+## Architecture
+
+```mermaid
+graph LR
+    subgraph "Providers"
+        Encrypted[Encrypted Storage<br/>AES-256-GCM]
+        OnePass[1Password CLI]
```

Comment:
The diagram shows "1Password CLI" but the actual implementation uses the **1Password SDK**.

```suggestion
        OnePass[1Password SDK]
```

This is consistent with the text documentation on line 43 which correctly states `github.com/1password/onepassword-sdk-go`.

Evidence: `pkg/secrets/1password.go` imports the SDK package